### PR TITLE
SCP-1880: Datums and Redeemers Cardano API envelopes

### PR DESCRIPTION
**Pull request by koslambrou**
Support for exporting Datums and Redeemers to Cardano API envelopes

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: plutus-ledger-api/src/Plutus/V1/Ledger/Api.hs [hunk 1]
```diff
@@ -105,6 +105,12 @@ import qualified UntypedPlutusCore.Evaluation.Machine.Cek         as UPLC
 plutusScriptEnvelopeType :: Text
 plutusScriptEnvelopeType = "PlutusV1Script"
 
+plutusDatumEnvelopeType :: Text
```
Comment (michaelpj) [inline plutus-ledger-api/src/Plutus/V1/Ledger/Api.hs:108]: A couple of things I'm not sure about here:
- Should these be `PlutusV1*` types? Datums and redeemers should be the same across versions, and maybe even across languages.
- Should these be here? If they're not plutus-specific they should probably be defined in `cardano-api`. I mentioned this in the issue. But perhaps we should just put something here until they are...

Comment (michaelpj) [reply plutus-ledger-api/src/Plutus/V1/Ledger/Api.hs:108]: I'll ask in #ledger

File: doc/plutus/howtos/exporting-a-script.rst [hunk 2]
```diff
@@ -13,6 +13,8 @@ For example, you might want to submit it as part of a manually created transacti
 Fortunately, it is quite simple to do this.
 You can either translate directly into CBOR using the `Serialise` typeclass from the `serialise` package, or you can create an envelope of the sort used by the Cardano node CLI.
 
+The same can also be done with `Datum` and `Redeemer` datatypes.
```
Comment (michaelpj) [inline doc/plutus/howtos/exporting-a-script.rst:16]: Maybe change the title of the doc since it includes more stuff now?

Comment (koslambrou) [reply doc/plutus/howtos/exporting-a-script.rst:16]: exporting-data? exporting-script-datum-redeemer? or other?

Comment (michaelpj) [reply doc/plutus/howtos/exporting-a-script.rst:16]: I meant the title at the top, the file name is okay. Maybe "How to export scripts, datums, and redeemers"

Comment (koslambrou) [reply doc/plutus/howtos/exporting-a-script.rst:16]: Done

**Review by michaelpj — APPROVED** [review]

**Pull request merged**